README.md

```diff
@@ -0,0 +1,136 @@
+# aws-assume-role
+
+This will get role credentials for you, managing 2FA devices, and set those
+credentials in environments. It stores the fetched credentials in Gnome Keyring
+or OSX Keychain so they are not readable from disk.
+
+## Install
+
+`gem install aws_assume_role`
+
+### Platform notes
+
+Gnome Keyring uses the [GirFFI](https://github.com/mvz/gir_ffi) bindings, which
+requires the introspection bindings to be installed (as well as gnome-keyring).
+`apt-get install gnome-keyring libgirepository1.0-dev` for Debian/Ubuntu.
+
+## Config file
+
+Create a config file, the default is `~/.aws/assume.yaml`
+
+```yaml
+---
+default:
+    set_environment: false
+    # credentials come from .aws/credentials default profile or environment
+
+scalefactory:
+    # if this profile is passed don't set the environment credentials
+    set_environment: false
+    # load credentials from sf_sso profile in .aws/credentials
+    profile: sf_sso
+
+
+# These use the scalefactory profile above
+xx_mgmt:
+    parent: scalefactory
+    set_environment: true
+    type: assume_role
+    region: eu-west-1
+    role_arn: arn:aws:iam::123456789012:role/RoleNameHere
+
+xx_test:
+    parent: scalefactory
+    set_environment: true
+    type: assume_role
+    role_arn: arn:aws:iam::123456789012:role/RoleNameHere
+
+xx:
+    type: list
+    set_environment: true
+    list:
+     - name:       xx_test
+       env_prefix: TEST_
+     - name:       xx_mgmt
+       env_prefix: MGMT_
+
+
+# These use the default above
+yy_mgmt:
+   set_environment: true
+   type: assume_role
+   role_arn: arn:aws:iam::123456789012:role/RoleNameHere
+
+yy_test:
+   set_environment: true
+   type: assume_role
+   role_arn: arn:aws:iam::123456789012:role/RoleNameHere
+
+xx:
+   type: list
+   set_environment: true
+   list:
+    - name:       xx_test
+      env_prefix: TEST_
+    - name:       xx_mgmt
+      env_prefix: MGMT_
+
+
+```
+
+I have the following entry in `~/.aws/credentials`:
+
+```
+[sf_sso]
+aws_access_key_id = 1234567890010
```

Comment:
Ok, will merge at present. I think there's a case for extending this, but ok to defer it.